Src/java/engine-fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/converter/Dstu2FhirTypeConverter.java

```diff
@@ -227,7 +228,35 @@ public IBase toFhirTuple(Tuple value) {
             return null;
         }
 
-        throw new NotImplementedException("can't convert Tuples");
+        var parameters = new Parameters();
+        if (value.getElements().isEmpty()) {
+            return parameters;
+        }
+
+        var param = parameters.addParameter();
+        for (String key : value.getElements().keySet()) {
+            var part = param.addPart();
+            part.setName(key);
+            var element = value.getElements().get(key);
+            if (element == null) {
+                part.addExtension()
```

Comment:
The data-absent-reason extension needs to be on the value, not the part:

```xml
<parameter>
  <extension url="http://hl7.org/fhir/StructureDefinition/cqf-cqlType">
    <valueString value="System.Boolean"/>
  </extension>
  <name value="Is pregnancy confirmed"/>
  <valueBoolean>
    <extension url="http://hl7.org/fhir/StructureDefinition/data-absent-reason">
      <valueCode value="unknown"/>
    </extension>
  </valueBoolean>
</parameter>
```

Reply:
CQL does not track any metadata about types as part of the execution. We do not even know what type the missing value is. All we know is that "X" is "null".

Reply:
I don't know what I was thinking. You're right.